sdk/monitor/Azure.Monitor.Query/src/Models/MonitorQueryModelFactory.cs

```diff
@@ -36,5 +36,35 @@ public static LogsQueryResult LogsQueryResult(IReadOnlyList<LogsTable> allTables
             JsonElement errorJson = error.ToObjectFromJson<JsonElement>();
             return new LogsQueryResult(allTables, statisticsJson, visualizationJson, errorJson);
         }
+
+        /// <summary> Initializes a new instance of LogsTableRow. </summary>
+        /// <returns> A new <see cref="Models.LogsTableColumn"/> instance for mocking. </returns>
+        public static LogsTableRow LogsTableRow(IReadOnlyList<LogsTableColumn> columns, Object[] rows)
+        {
+            Dictionary<string, int> columnDictionary = new();
+
+            for (var index = 0; index < columns.Count; index++)
+            {
+                columnDictionary[columns[index].Name] = index;
+            }
+
+            JsonElement row = JsonElementFromObject(rows);
+            return new LogsTableRow(columnDictionary, columns, row);
+        }
+
+        /// <summary> Initializes a new instance of LogsTable. </summary>
+        /// <exception cref="ArgumentNullException"> <paramref name="name"/> or <paramref name="rows"/> is null. </exception>
+        public static LogsTable LogsTable(string name, IEnumerable<LogsTableRow> rows, IEnumerable<LogsTableColumn> columns)
```

Comment:
Nit - maybe reorder the parameters so `columns` comes before `rows` since that appears to be more common and aligned with the service's conceptual model.